github/repos_contents.go

```diff
@@ -267,9 +267,13 @@ func (s *RepositoriesService) GetArchiveLink(owner, repo string, archiveformat a
 	} else {
 		resp, err = s.client.client.Transport.RoundTrip(req)
 	}
-	if err != nil || resp.StatusCode != http.StatusFound {
-		return nil, newResponse(resp), err
+	if err != nil {
+		return nil, nil, err
+	}
+	resp.Body.Close()
+	if resp.StatusCode != http.StatusFound {
+		return nil, newResponse(resp), fmt.Errorf("unexpected status code: %s", resp.Status)
 	}
 	parsedURL, err := url.Parse(resp.Header.Get("Location"))
-	return parsedURL, newResponse(resp), err
+	return parsedURL, newResponse(resp), nil
```

Comment:
This line was actually correct before. You should undo the change here. It returns `err` from the line above:

```Go
parsedURL, err := url.Parse(resp.Header.Get("Location"))
```

And that `err` may be nil or not nil.